src/components/common/DropdownButton.js

```diff
@@ -1,77 +1,82 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React from 'react';
-import {
-  Dropdown, DropdownToggle, DropdownMenu, DropdownItem,
-} from 'reactstrap';
+import React, { useState } from 'react';
+import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
 
 /** --------Props---------------
- * dropdownItems: array of strings, each string being the name of a Program
+ * children: array of strings, each string being the name of a Program
  * displayValue: string to be displayed as the placeholder for the dropdown
  * onSelect: function called when an item is selected in the dropdown
  */
 /** --------Optional props--------
  * defaultOpen: boolean determining if the dropdown should start off open or closed
  */
-export default class DropdownButton extends React.Component {
-  constructor(props) {
-    super(props);
 
-    this.state = {
-      dropdownOpen: this.props.defaultOpen || false,
-    };
-  }
+const DropdownButton = (props) => {
+  const {
+    dirty,
+    onSelect,
+    children,
+    defaultOpen,
+    displayValue,
+    displayClass,
+    toggleClass,
+    toggleColor,
+    toggleSize,
+  } = props;
 
-  //= =============React Lifecycle Functions===================//
-  componentDidMount() {}
+  const parentDisplay = displayClass + '-language-dropdown';
+  const itemDisplay = displayClass + '-language-dropdown-closed-content';
+  const [dropdownOpen, setdropdownOpen] = useState(defaultOpen || false);
 
-  toggleHandler = (prevVal) => {
-    this.setState({ dropdownOpen: !prevVal });
+  const toggleHandler = (prevVal) => {
+    setdropdownOpen(!prevVal);
   };
 
-  selectLanguage = (program) => {
-    let result = true;
-    if (this.props.dirty) {
-      result = window.confirm('Are you sure you want to change programs? You have unsaved changes');
-    }
+  const renderDropdownItems = () =>
+    children.map(({ display, value }) => {
```

Comment:
Just to clarify, for "children" it would be alright if I processed it outside and children was already just an array containing all the icons and display names etc?

Reply:
in general, `children` refers to the components inside of the component which `children` or `child` is passed to.
so children would be something more used like
```
<DropdownButton>
   <child>asdfasdf</child>
</DropdownButton>
```
as per what timothy said in a different comment!

so with that, if it was just pure JSX, then yes, calling it "children" would be accurate